src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -229,80 +273,6 @@ public bool ShouldCollapseOnOpen(string? filePath, BlockStructureOptions blockSt
         return new SymbolMappingResult(project, resolutionResult.Symbol);
     }
 
-    public void CleanupGeneratedFiles()
-    {
-        using (_gate.DisposableWait())
-        {
-            // Release our mutex to indicate we're no longer using our directory and reset state
-            if (_mutex != null)
-            {
-                _mutex.Dispose();
```

Comment:
no need to dispose our own mutex.  when VS shuts down that will automatically happen.